app/src/main/java/app/revanced/integrations/youtube/patches/components/HideKeywordContentFilter.java

```diff
@@ -0,0 +1,137 @@
+ * - Searching for a keyword phrase will give no search results
+ * - Filtering a channel name can still show Shorts from that channel in the search results
+ * - Some layout component residue will remain, such as the video chapter previews for some search results.
+ * - Keywords are case sensitive, but some casing variation is manually added.
+ *   (such as "Mr Beast" automatically filtering "mr beast" and "MR BEAST").
+ */
+@SuppressWarnings("unused")
+@RequiresApi(api = Build.VERSION_CODES.N)
+final class HideKeywordContentFilter extends Filter {
+
+    /**
+     * Substrings that are always first in the path.
+     */
+    private final StringFilterGroup startsWithFilter = new StringFilterGroup(
+            Settings.HIDE_KEYWORD_CONTENT,
+            "home_video_with_content.eml",
+            "search_video_with_context.eml",
+            "related_video_with_content",
+            "inline_shorts",
+            "shorts_video_cell",
+            "shorts_pivot_item.eml");
+
+    /**
+     * Substrings that are never at the start of the path.
+     */
+    private final StringFilterGroup containsFilter = new StringFilterGroup(
+            Settings.HIDE_KEYWORD_CONTENT,
+            "modern_type_shelf_header_content.eml");
+
+    private final ByteTrieSearch bufferSearch = new ByteTrieSearch();
+
+    /**
+     * Minimum keyword/phrase length to prevent excessively broad content filtering.
+     */
+    private static final int MINIMUM_KEYWORD_LENGTH = 3;
+
+    private static List<String> splitAndVerifyStrings(StringSetting setting) {
+        String[] split = setting.get().split("\n");
+        List<String> keywords = new ArrayList<>(split.length);
+
+        for (String phrase : split) {
+            // Remove any trailing white space the user may have accidentally included.
+            phrase = phrase.stripTrailing();
+            if (phrase.isBlank()) continue;
+
+            if (phrase.length() < MINIMUM_KEYWORD_LENGTH) {
+                Utils.showToastLong(str("revanced_hide_keyword_toast_invalid_length", MINIMUM_KEYWORD_LENGTH, phrase));
+                Utils.showToastLong(str("revanced_hide_keyword_toast_reset"));
+                setting.resetToDefault();
+                return splitAndVerifyStrings(setting);
+            }
+            keywords.add(phrase);
+        }
+
+        return keywords;
+    }
+
+    private void parseKeywords() {
+        List<String> keywords = splitAndVerifyStrings(Settings.HIDE_KEYWORD_CONTENT_PHRASES);
+        if (keywords.isEmpty()) {
+            return;
+        }
+
+        // Add common casing that might appear.
+        // If a desired keyword has mixed casing (such as "Matt Whatever")
+        // the user will need to add that exact string.
+        // This could be simplified by adding case insensitive search to the prefix search,
+        // but that also brings a small performance hit.
+        List<String> modifiedKeywords = new ArrayList<>();
+        for (String keyword : keywords) {
```

Comment:
Are these hardcoded behaviors of the content filter known to the user? In the description or somewhere, quirky behaviours such as the minimum limit of 3 characters, the casing as well as the locations in which the filter is active and other kinds of behaviors must be known to the user.

Reply:
> The 3 character minimum is not explained in the settings, but they will find out when they enter a 1 or 2 character string (it shows an error toast on app startup).

The problem with this is, that it will nuke their filters and that may be annoying as at the point they see their error, their filters would be gone and they'd need to redo them.